affordance_primitives/test/test_task_monitor.cpp

```diff
@@ -0,0 +1,194 @@
+///////////////////////////////////////////////////////////////////////////////
+//      Title     : test_task_monitor.cpp
+//      Project   : affordance_primitives
+//      Created   : 02/28/2022
+//      Author    : Adam Pettinger
+//      Copyright : Copyright© The University of Texas at Austin, 2014-2022. All
+//      rights reserved.
+//
+//          All files within this directory are subject to the following, unless
+//          an alternative license is explicitly included within the text of
+//          each file.
+//
+//          This software and documentation constitute an unpublished work
+//          and contain valuable trade secrets and proprietary information
+//          belonging to the University. None of the foregoing material may be
+//          copied or duplicated or disclosed without the express, written
+//          permission of the University. THE UNIVERSITY EXPRESSLY DISCLAIMS ANY
+//          AND ALL WARRANTIES CONCERNING THIS SOFTWARE AND DOCUMENTATION,
+//          INCLUDING ANY WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
+//          PARTICULAR PURPOSE, AND WARRANTIES OF PERFORMANCE, AND ANY WARRANTY
+//          THAT MIGHT OTHERWISE ARISE FROM COURSE OF DEALING OR USAGE OF TRADE.
+//          NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH RESPECT TO THE USE OF
+//          THE SOFTWARE OR DOCUMENTATION. Under no circumstances shall the
+//          University be liable for incidental, special, indirect, direct or
+//          consequential damages or loss of profits, interruption of business,
+//          or related expenses which may arise from use of software or
+//          documentation, including but not limited to those resulting from
+//          defects in software and/or documentation, or loss or inaccuracy of
+//          data of any kind.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+#include <affordance_primitives/task_monitor/task_monitor.hpp>
+#include <affordance_primitives/msg_types.hpp>
+
+#include <thread>
+
+// constexpr double EPSILON = 1e-4;
+
+using affordance_primitives::ExecutionResult;
+
+TEST(TaskMonitor, start_stop_monitor)
+{
+  ros::NodeHandle nh;
+
+  // Default parameters
+  affordance_primitives::AffordanceParameter params;
+
+  // Create and start monitor with no timeout
+  affordance_primitives::TaskMonitor monitor(nh, "ft_topic_name");
+  ASSERT_NO_THROW(monitor.startMonitor(params));
+
+  // Wait a sec, then stop monitor
+  ros::Duration(0.25).sleep();
```

Comment:
Just highlighting that we are waiting for 0.25 seconds, and not a second, based on the comment